File: src/C3.Catalogue/CassetteModels/CassetteModel.vb
Namespace CassetteModels

    Public NotInheritable Class CassetteModel

        Public Sub New(
                brandCode As String,
                typeNumber As Integer,
                modelName As String,
                code As String,
                identifier As String,
                displayName As String,
                tapeCount As Integer,
                addedAt As DateTime,
                notes As String)

            Me.BrandCode = brandCode
            Me.TypeNumber = typeNumber
            Me.ModelName = modelName
            Me.Code = code
            Me.Identifier = identifier
            Me.DisplayName = displayName
            Me.TapeCount = tapeCount
            Me.AddedAt = addedAt
            Me.Notes = notes
        End Sub

        Public ReadOnly Property BrandCode As String
        Public ReadOnly Property TypeNumber As Integer
        Public ReadOnly Property ModelName As String
        Public ReadOnly Property Code As String
        Public ReadOnly Property Identifier As String
        Public ReadOnly Property DisplayName As String
        Public ReadOnly Property TapeCount As Integer
        Public ReadOnly Property AddedAt As DateTime
        Public ReadOnly Property Notes As String

    End Class

End Namespace

File: src/C3.Catalogue/CassetteModels/CassetteModel.cs
using System;

namespace C3.Catalogue.CassetteModels
{
    public sealed class CassetteModel
    {
        public CassetteModel(
            string brandCode,
            int typeNumber,
            string modelName,
            string code,
            string identifier,
            string displayName,
            int tapeCount,
            DateTime addedAt,
            string notes)
        {
            BrandCode = brandCode;
            TypeNumber = typeNumber;
            ModelName = modelName;
            Code = code;
            Identifier = identifier;
            DisplayName = displayName;
            TapeCount = tapeCount;
            AddedAt = addedAt;
            Notes = notes;
        }

        public string BrandCode { get; }

        public int TypeNumber { get; }

        public string ModelName { get; }

        public string Code { get; }

        public string Identifier { get; }

        public string DisplayName { get; }

        public int TapeCount { get; }

        public DateTime AddedAt { get; }

        public string Notes { get; }
    }
}
